Forge Standard Library docs (#207)
foundry-rs/book: src/forge/writing-tests.md

Tests are written in Solidity. If the test function reverts, the test fails, otherwise it passes.

> Using the [Forge Standard Library](https://github.com/foundry-rs/forge-std) is the preferred way of writing tests with Foundry.

Reviewer: ```suggestion
Let's go over the most common way of writing tests, using the [Forge Standard Library](https://github.com/foundry-rs/forge-std)'s `Test` contract, which is the preferred way of writing tests with Forge.
```

Install it by running:

```bash
forge install foundry-rs/forge-std
```

> In this section, we'll go over the basics using the functions from the [Dappsys Test](https://github.com/dapphub/ds-test) library, which is included in the Forge Standard Library. You will learn how to use more advanced stuff from the Forge Standard Library [soon](./forge-std.md). 

Reviewer: ```suggestion
In this section, we'll go over the basics using the functions from the Forge Std's `Test` contract, which is itself a superset of [DSTest](https://github.com/dapphub/ds-test). You will learn how to use more advanced stuff from the Forge Standard Library [soon](./forge-std.md). 
```

> Dappsys Test provides basic logging and assertion functionality. To get access to the functions, import `forge-std/Test.sol` and inherit from `Test` in your test contract:

Reviewer: ```suggestion
DSTest provides basic logging and assertion functionality. To get access to the functions, import `forge-std/Test.sol` and inherit from `Test` in your test contract:
```

    ```solidity
    {{#include ../../projects/writing_tests/test/Basic.t.sol:testFailSubtract43}}
    ```

>     A good practice is to use something like `testCannot` in combination with the [`expectRevert`](../cheatcodes/expect-revert.md) cheatcode (cheatcodes are explained in greater detail in the following [section](./cheatcodes.md)).
>     <br>
>     Now, instead of using `testFail`, you know exactly what reverted:

Reviewer: I think this should work (double space after a line adds a newline)

```suggestion
    A good practice is to use something like `testCannot` in combination with the [`expectRevert`](../cheatcodes/expect-revert.md) cheatcode (cheatcodes are explained in greater detail in the following [section](./cheatcodes.md)).  

    Now, instead of using `testFail`, you know exactly what reverted:
```
